src/Entity/EntityIdValue.php

```diff
@@ -31,8 +31,8 @@ public function __construct( EntityId $entityId ) {
 	 */
 	public function serialize() {
 		return json_encode( [
-			$this->entityId->getEntityType(),
-			$this->getNumericId()
+			get_class( $this->entityId ),
```

Comment:
My remark is not about prettiness, it is about security.

> But all we use this for is cloning statements.

That does no reassure me. Kinda like saying "no need to escape this SQL argument since right now only we call this function with non-user-input".

> So, having the choice to try and access some kind of entity id type registry via a global, or putting the class name into the serialization, I opted for the latter.

Why not just (de)serialize the ID?